CONTRIBUTING.md

```diff
@@ -4,7 +4,56 @@ Contributing to OpenWeave
 Want to contribute? Great! First, read this page (including the small
 print at the end).
 
-### Before you contribute ###
+## Bugs
+
+If you find a bug in the source code, you can help us by [submitting a GitHub Issue](https://github.com/openweave/openweave-core/issues/new).  The best bug reports provide a detailed description of the issue and step-by-step instructions for predictably reproducing the issue.  Even better, you can [submit a Pull Request](#submitting-a-pull-request) with a fix.
+
+## New Features
+
+You can request a new feature by [submitting a GitHub Issue](https://github.com/openweave/openweave-core/issues/new).
+
+If you would like to implement a new feature, please consider the scope of the new feature:
+
+* *Large feature*: first [submit a GitHub
+  Issue](https://github.com/openweave/openweave-core/issues/new) and communicate
+  your proposal so that the community can review and provide feedback.  Getting
+  early feedback will help ensure your implementation work is accepted by the
+  community.  This will also allow us to better coordinate our efforts and
+  minimize duplicated effort.
+
+* *Small feature*: can be implemented and directly [submitted as a Pull
+  Request](#submitting-a-pull-request).
+
+## Contributing Code
+
+The OpenWeave-Core follows the "Fork-and-Pull" model for accepting contributions.
+
+### Initial Setup
+
+Setup your GitHub fork and continuous-integration services:
+
+1. Fork the [OpenWeave-Core
+   repository](https://github.com/openweave/openweave-core) by clicking "Fork"
+   on the web UI.
+
+2. Enable [Travis CI](https://travis-ci.org/) and
+   [AppVeyor](https://ci.appveyor.com/) by logging in the respective services
```

Comment:
We don't use AppVeyor on OpenWeave to my knowledge.

Reply:
Done.